### ISSUE TITLE
bug: <title>Reproduce issue

### ISSUE THREAD

**Issue by Harry-mic** [axon-rl/gem#78]
### Describe the bug

Hi, thanks for this awesome work!
I encounter a reproduce issue that the performance I reproduce on some tasks(such as GuessTheNumber, MineSweeper, Mastermind) doesn't reach the performance in the blog: https://axon-rl.notion.site/gem#238dba9c0091800bb5eedaf66517d840.

Here is the reproduced performance of GuessTheNumber:
<img width="406" height="319" alt="image" src="https://github.com/user-attachments/assets/4633c222-0f2c-48d0-a1af-8978b2e5d58e" />  
<img width="183" height="116" alt="image" src="https://github.com/user-attachments/assets/c22d825d-9464-4001-8493-ccf512492da3" />

MineSweeper:
<img width="280" height="324" alt="image" src="https://github.com/user-attachments/assets/44cd4a20-b929-44b9-a73b-f12ed90135f0" />
<img width="152" height="119" alt="image" src="https://github.com/user-attachments/assets/93eb944d-67f9-4e52-b8ac-abaac5b13c07" />

Mastermind:
<img width="277" height="319" alt="image" src="https://github.com/user-attachments/assets/fc27c6b1-0ed6-490e-bb35-992505030ab7" />
<img width="162" height="119" alt="image" src="https://github.com/user-attachments/assets/b3b1724f-eefe-4edc-b6f9-b898fd605ec3" />

The parameters I used is :

```
python examples/train_oat.py \
    --env_id game:**(three game )** \
    --prompt_template qwen3_game \
    --wrappers concat \
    --gamma 0.9 \
    --norm_return \
    --gpus 8 \
    --gradient-checkpointing \
    --num_samples 1 \
    --rollout_batch_size 128 \
    --num_env 2 \
    --rollout_batch_size_per_device 16 \
    --pi_buffer_maxlen_per_device 16 \
    --pretrain  Qwen3-1.7B-Base \
    --enable_prefix_caching \
    --collocate \
    --vllm_sleep \
    --vllm_gpu_ratio 0.45 \
    --rnd-seed \
    --learning_rate 0.000001 \
    --lr_scheduler constant \
    --lr_warmup_ratio 0 \
    --num_ppo_epochs 2 \
    --train_batch_size 128 \
    --train_batch_size_per_device 1 \
    --beta 0 \
    --max_model_len 12800 \
    --generate_max_length 4096 \
    --temperature 1.0 \
    --top_p 1 \
    --eval_steps -1 \
    --save_steps 450 \
    --eval_temperature 0.6 \
    --eval_top_p 0.95 \
    --eval_generate_max_length 4096 \
    --max_train 65000 \
    --max_save_num 1 \
    --use-wb \
    --wb-run-name oat-qwen3-1.7b-base-game:**(three game )** \
    --wb_project gem
```

### To Reproduce

see above

### The gem version

the newest version

**Comment by lkevinzc** [comment]
Hi @Harry-mic , thanks for your interest!

For multi-turn tasks, typically we look at the return (accumulated over all turns) instead of per-step reward. The figure below is from my run:

<img width="1374" height="610" alt="Image" src="https://github.com/user-attachments/assets/4be2fbbd-419b-49a3-bed4-5f4e53e816e1" />

We can see that the average per-step reward is about 0.3 (same as yours), but the episode return is aligned with the curve in the blog.

**Comment by Harry-mic** [comment]
Thanks for this explanation! The issue is solved, so i close the comment.
